### PR TITLE
PIM-9499: Fix warning display when a job is running with warnings

### PR DESCRIPTION
**Pull request by anaisbaunelemaire**
**Description (for Contributor and Core Developer)**

When a job was running, like a product import, when there were warnings displayed, it was impossible to read them as every time we clicked on "display item", the item displayed immediately closed back.

Confirm by PM: "when a user expands a warning, it has to stay expanded until he decides to collapse."

The problem was that the page was rendered after each display of a new warning.
So I created an expandedWarnings object in which the warnings are stored. 
Thus, we can click on "display item" and view the details of the warning message even during the execution of a job.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/job/execution/summary-table.js [hunk 1]
```diff
@@ -28,10 +29,16 @@ define(
...
+                if (this.expandedWarnings[stepIndex]) {
+                    if (this.expandedWarnings[stepIndex].includes(parseInt(warningIndex))) {
+                        const warningIndexToRemove = this.expandedWarnings[stepIndex].indexOf(parseInt(warningIndex));
+                        this.expandedWarnings[stepIndex].splice(warningIndexToRemove, 1);
+                    } else {
+                        this.expandedWarnings[stepIndex].push(parseInt(warningIndex));
+                    }
+                } else {
+                    this.expandedWarnings[stepIndex] = [parseInt(warningIndex)];
+                }
```
Comment (juliensnz) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/job/execution/summary-table.js:41]: What do you think about this solution:

```suggestion
                const stepIndex = parseInt(event.currentTarget.dataset.stepIndex);
                const warningIndex = parseInt(event.currentTarget.dataset.warningIndex);
                
                this.expandedWarnings[stepIndex] = this.expandedWarnings[stepIndex] || [];
                
                this.expandedWarnings[stepIndex] = this.expandedWarnings[stepIndex].includes(warningIndex) ?
                    this.expandedWarnings[stepIndex].filter(index => index === warningIndex) :
                    [...this.expandedWarnings[stepIndex], warningIndex];
```

Comment (anaisbaunelemaire) [reply src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/job/execution/summary-table.js:41]: Many Thanks for your feedback, Julien !
Your proposal seemed good to me but I tested and the "hide item" doesn't work during the execution of the job.
Another problem: once the execution of the job is finished, when I have 2 open warnings and I want to close one, the other closes. I'll work on your piece of code, to find out where the error is.

File: src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/job/execution/summary-table.js [hunk 2]
```diff
@@ -18,19 +18,21 @@ define(['jquery', 'underscore', 'oro/translator', 'pim/form', 'pim/template/job-
...
-      model.stepExecutions[stepIndex].warnings[warningIndex].expanded = !model.stepExecutions[stepIndex].warnings[
-        warningIndex
-      ].expanded;
+      const stepIndex = parseInt(event.currentTarget.dataset.stepIndex);
+      const warningIndex = parseInt(event.currentTarget.dataset.warningIndex);
+
+      this.expandedWarnings[stepIndex] = this.expandedWarnings[stepIndex] || [];
 
+      this.expandedWarnings[stepIndex] = this.expandedWarnings[stepIndex].includes(warningIndex)
+        ? this.expandedWarnings[stepIndex].filter(index => index !== warningIndex)
```
Comment (anaisbaunelemaire) [inline src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/job/execution/summary-table.js:34]: @juliensnz the mistake was there:
`index !== warningIndex` (et non `index === warningIndex`) 
It works, now ! Thx for your improvement proposal :-)

File: CHANGELOG.md [hunk 3]
```diff
@@ -73,15 +74,13 @@
 - PIM-9371: Disable save button when user creation form is not ready to submit
 - RAC-178: When launching a job, the notification contains a link to the job status
 - PIM-9485: Change ACL name “Remove a product model” to “Remove a product model (including children)”
-- BH-138: clear Locale cache on save
```
Comment (jmleroux) [inline CHANGELOG.md:76]: ???

**Review by jmleroux — CHANGES_REQUESTED** [review]
The CHANGELOG should not remoive entries

File: CHANGELOG.md (hunk 3 above)
Comment (nmarniesse) [inline CHANGELOG.md:76]: Why removing this line and the two others below?

**Review by nmarniesse — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (anaisbaunelemaire) [reply CHANGELOG.md:76]: I don't understand !? I resolved the conflicts on the changelog file accepting master and local changes.
Well, I'll revert that...!

Comment (anaisbaunelemaire) [reply CHANGELOG.md]: Solved @jmleroux

Comment (anaisbaunelemaire) [reply CHANGELOG.md]: Solved @nmarniesse ;-)

**Review by jmleroux — APPROVED** [review]

**Pull request merged**